crates/cubecl-cpu/src/runtime.rs

```diff
@@ -0,0 +1,117 @@
+use cubecl_core::{
+    CubeDim, DeviceId, MemoryConfiguration, Runtime, channel::MutexComputeChannel,
+    client::ComputeClient,
+};
+use cubecl_runtime::{
+    ComputeRuntime, DeviceProperties,
+    memory_management::{HardwareProperties, MemoryDeviceProperties, MemoryManagement},
+    storage::{BytesStorage, ComputeStorage},
+};
+use sysinfo::System;
+
+use crate::{
+    compiler::{MLIRCompiler, mlir::register_supported_types},
+    compute::server::{CpuContext, CpuServer},
+    device::CpuDevice,
+};
+
+#[derive(Default)]
+pub struct RuntimeOptions {
+    /// Configures the memory management.
+    pub memory_config: MemoryConfiguration,
+}
+
+#[derive(Debug)]
+pub struct CpuRuntime;
+
+static RUNTIME: ComputeRuntime<CpuDevice, Server, Channel> = ComputeRuntime::new();
+
+pub type CpuCompiler = MLIRCompiler;
+
+type Server = CpuServer;
+type Channel = MutexComputeChannel<Server>;
+
+fn create_client(options: RuntimeOptions) -> ComputeClient<Server, Channel> {
+    let max_thread = std::thread::available_parallelism()
+        .map(|n| n.get() as u32)
+        .unwrap_or(1);
+    // TODO determine how parallelism should be distributed between cube_count and cube_dim
+    // For now cube_count is parallel and cube_dim sequential
```

Comment:
It is the opposite, `cube_dim` needs be parallel and `cube_count` can be whatever.